src/Analyzers/CSharp/Tests/UseCollectionInitializer/UseCollectionInitializerTests_CollectionExpression.cs

```diff
@@ -0,0 +1,1772 @@
+
+            class C
+            {
+                void M()
+                {
+                    List<Action> list1 = [|new|] List<Action>();
+                    [|list1.Add(|]() => {
+                        List<int> list2 = [|new|] List<int>();
+                        [|list2.Add(|]2);
+                    });
+                }
+            }
+            """,
+            FixedCode =
+            """
+            using System;
+            using System.Collections.Generic;
+
+            class C
+            {
+                void M()
+                {
+                    List<Action> list1 = [() =>
+                    {
+                        List<int> list2 = [2];
+                    }
+
+                    ];
+                }
+            }
+            """,
+            BatchFixedCode =
+            """
+            using System;
+            using System.Collections.Generic;
+
+            class C
+            {
+                void M()
+                {
+                    List<Action> list1 = [() =>
+                    {
+                        List<int> list2 = [2];
+                    }
+
+                    ];
+                }
+            }
+            """,
+            LanguageVersion = LanguageVersion.Preview,
+        }.RunAsync();
+    }
+
+    [Fact]
+    public async Task TestTrivia1()
+    {
+        await TestInRegularAndScriptAsync(
+            """
+            using System.Collections.Generic;
+            class C
+            {
+                void M()
+                {
+                    List<int> c = [|new|] List<int>();
+                    [|c.Add(|]1); // Goo
+                    [|c.Add(|]2); // Bar
+                }
+            }
+            """,
+            """
+            using System.Collections.Generic;
+            class C
+            {
+                void M()
+                {
+                    List<int> c = [1, // Goo
+                        2 // Bar
+            ];
+                }
+            }
+            """);
+    }
+
+    [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseObjectInitializer)]
+    [WorkItem("https://github.com/dotnet/roslyn/issues/46670")]
+    public async Task TestTriviaRemoveLeadingBlankLinesForFirstElement()
+    {
+        await TestInRegularAndScriptAsync(
+            """
+            using System.Collections.Generic;
+            class C
+            {
+                void M()
+                {
+                    List<int> c = [|new|] List<int>();
+
+                    // Goo
+                    [|c.Add(|]1);
+
+                    // Bar
+                    [|c.Add(|]2);
+                }
+            }
+            """,
+            """
+            using System.Collections.Generic;
+            class C
+            {
+                void M()
+                {
+                    List<int> c = [
+                    // Goo
+                    1,
+                        // Bar
+                        2];
```

Comment:
note: some formatting here is *not good*.  I tried getting this better, but gave up on the formatting engine side.  will need help there to get this first class.